File: backend/app/schemas/metrics.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class MetricSnapshot(BaseModel):
    cpu_percent: float = 0
    memory_percent: float = 0
    memory_used: int = 0
    memory_total: int = 0
    disk_percent: float = 0
    disk_read_bytes: int = 0
    disk_write_bytes: int = 0
    net_bytes_sent: int = 0
    net_bytes_recv: int = 0
    load_1m: float = 0
    load_5m: float = 0
    load_15m: float = 0
    process_count: int = 0
    uptime_seconds: int = 0


class MetricHistoryPoint(BaseModel):
    time: str
    value: float


class MetricHistoryResponse(BaseModel):
    metric: str
    range: str
    data: list[MetricHistoryPoint]

File: frontend/src/lib/api/metrics.ts
import { apiFetch } from './client';

export interface MetricSnapshot {
	cpu_percent: number;
	memory_percent: number;
	memory_used: number;
	memory_total: number;
	disk_percent: number;
	net_bytes_sent: number;
	net_bytes_recv: number;
	load_1m: number;
	load_5m: number;
	load_15m: number;
	process_count: number;
	uptime_seconds: number;
}

export interface MetricHistoryPoint {
	time: string;
	value: number;
}

export interface MetricHistoryResponse {
	metric: string;
	range: string;
	data: MetricHistoryPoint[];
}

export async function getMetricsSummary(): Promise<MetricSnapshot> {
	return apiFetch<MetricSnapshot>('/metrics/summary');
}

export async function getMetricsHistory(metric: string, range = '1h'): Promise<MetricHistoryResponse> {
	return apiFetch<MetricHistoryResponse>(`/metrics/history?metric=${metric}&range=${range}`);
}
